added optional argument to the assume role for a custome duration

detnon opened the pull request:
The session credentials are timing out before it finishes which also means it can’t write the statefile when The windows sandbox terraform is being ran

detnon committed:
added optional argument to the assume role for a custome duration

danjoneslf commented on bin/sts-assume-role.sh:
```diff
@@ -10,10 +10,11 @@ fi
 
 role_arn="$1"
 region="$2"
+duration"$3"
```
There's a missing = here. I think I'd probably handle the default here and have `duration=${3:=1800}` using the := instead of :-. They do similar things but :- makes it look like the default is -1800s.

detnon committed:
fixed duration argument

danjoneslf commented on bin/sts-assume-role.sh:
```diff
@@ -13,7 +13,7 @@ region="$2"
                     --role-arn "${role_arn}" \
                     --role-session-name "concourse-task" \
-                    --duration 1800)
+                    --duration ${3:=1800})
```
That's not what I meant. 

What I meant was keep the duration variable where it was but handle the default replacement there between lines 12 and 13. `:=` does assignment so `duration=${3:=1800}` should result in $duration set to $3 if it exists or 1800 if not which you can then pass to the `aws sts` call. 

So instead of doing the default replacement when you call `aws sts` you do the default replacement when you read the input arguments.

detnon committed:
fixed duration argument

danjoneslf committed:
Handle default duration and region at top of assume script

I would prefer to use `:=` because for the duration `:-` looks
like you're assigning a negative duration but unfortunately you
can't overwrite the argument values so you can't assign to `$3`

Handling the defaults immediately at the top of the script makes
it clearer what's happening if you don't specify an input arg.

danjoneslf reviewed (approved):
I've made the changes to this myself. 

I'd have preferred to use `:=` but you're not allowed to overwrite the input arg values so I've had to go back to `:-`.
